### PR TITLE
issue #4800 feat(nimbus): expose latest review approval in GQL

### PR DESCRIPTION
**Pull request by lmorchard**
Because:

- we want the review flow UI to properly indicate a locally approved
  experiment that is still pending approval in Remote Settings

This commit:

- adds review_approval field to GQL experiment queries

- adds latest_review_approval method to NimbusChangeLogManager

- tweaks latest_review_request method to include transition from
  PREVIEW/IDLE to DRAFT/REVIEW

- tweaks can_review to report true also while in WAITING publish_status

- adds publish_status to NimbusExperiment list in admin

### CODE REVIEW

**Comment by lmorchard** [comment]
Starting to dig into issue #4800 and found a need to expose this in order to enable this UI case when the experiment is approved locally but pending approval in Remote Settings:

https://storage.googleapis.com/mozilla-storybooks-experimenter/commits/ba828eef86092c0cb5b77a7579f3c6bcef68fdfe/nimbus-ui/index.html?path=/story/pages-requestreview-exp-1143-end-experiment-review-refactor--review-pending-in-remote-rettings-user-can-review

File: app/experimenter/experiments/models/nimbus.py
Comment (LZoog) [inline app/experimenter/experiments/models/nimbus.py:443]: I had to think this through for a sec since when the experiment is being reviewed it'll be in draft, but this seems right since IIUC a review request can _occur_ when it's in `PREVIEW`.

**Review by LZoog — APPROVED** [review]
This LGTM, though might be worth having @jaredlockhart glance through this before merging.

File: app/experimenter/experiments/models/nimbus.py
Comment (jaredlockhart) [inline app/experimenter/experiments/models/nimbus.py:275]: Why change this?  When it's in waiting it can no longer be reviewed?

File: app/experimenter/experiments/models/nimbus.py
Comment (jaredlockhart) [reply app/experimenter/experiments/models/nimbus.py:443]: Actually we might be able to remove this part of the filter altogether, now that we have publish status we don't have to look at transitions of status>status to look for review requests, this might be sufficient:

```py
                old_publish_status=NimbusExperiment.PublishStatus.IDLE,
                new_status=NimbusExperiment.Status.DRAFT,
                new_publish_status=NimbusExperiment.PublishStatus.REVIEW,
```

**Comment by LZoog** [comment]
Oh oops, I was sitting too long on the "files changed" page and didn't see @jaredlockhart beat me to a review by a few minutes😆

File: app/experimenter/experiments/models/nimbus.py
Comment (LZoog) [reply app/experimenter/experiments/models/nimbus.py:275]: Doesn't this change cover [this state](https://storage.googleapis.com/mozilla-storybooks-experimenter/commits/ba828eef86092c0cb5b77a7579f3c6bcef68fdfe/nimbus-ui/index.html?path=/story/pages-requestreview-exp-1143-end-experiment-review-refactor--review-pending-in-remote-rettings-user-can-review); isn't it in `WAITING` there?

Comment (jaredlockhart) [reply app/experimenter/experiments/models/nimbus.py:275]: Yeah but you can't review from experimenter there, that's already done.

Comment (lmorchard) [reply app/experimenter/experiments/models/nimbus.py:275]: It *can* be reviewed, though - in Remote Settings: https://storage.googleapis.com/mozilla-storybooks-experimenter/commits/ba828eef86092c0cb5b77a7579f3c6bcef68fdfe/nimbus-ui/index.html?path=/story/pages-requestreview-exp-1143-end-experiment-review-refactor--review-pending-in-remote-rettings-user-can-review

Comment (lmorchard) [reply app/experimenter/experiments/models/nimbus.py:275]: We're using the `canReview` flag to display that pending-in-remote-settings UI

Comment (jaredlockhart) [reply app/experimenter/experiments/models/nimbus.py:275]: Oh I see you're gating showing the RS link on can_review because the owner can't do that either. Okay yep that makes sense 👍

File: app/experimenter/experiments/admin/nimbus.py
Comment (jaredlockhart) [inline app/experimenter/experiments/admin/nimbus.py:82]: Yeah I'm guilty of sneaking admin changes like this into other PRs too.  Only problem with it is sometimes you end up having to back a PR out for some reason and then forget 'wait I thought I added this to the admin....'

**Review by jaredlockhart — APPROVED** [review]
Yep looks good.  Can probably take that old_status filter out if you want 👍

File: app/experimenter/experiments/admin/nimbus.py
Comment (lmorchard) [reply app/experimenter/experiments/admin/nimbus.py:82]: I could split that out in to a separate PR if need be. It turned out useful to verify what this PR was doing along with [the not-yet-PR'd UI work](https://github.com/mozilla/experimenter/compare/4800-request-review-ui) I've got in progress

File: app/experimenter/experiments/models/nimbus.py
Comment (jaredlockhart) [reply app/experimenter/experiments/models/nimbus.py:275]: That slightly changes the intention of the method here:

https://github.com/mozilla/experimenter/blob/a892d6fbead11126894beac7f9b5ec665f75a92c/app/experimenter/experiments/api/v5/serializers.py#L356-L360

But maybe the publish_status transition filtering will be sufficient since the user can't move it out of waiting only the tasks can do that.

File: app/experimenter/experiments/admin/nimbus.py
Comment (jaredlockhart) [reply app/experimenter/experiments/admin/nimbus.py:82]: Yeah totally we also need to add a couple other things to the admin as well.  Actually yeah if you wanna break it out I can tack on the other ones I need too.

**Comment by lmorchard** [comment]
Gonna take the two thumbs-up and merge, though. More work to come for issue #4800, hopefully today

File: app/experimenter/experiments/admin/nimbus.py
Comment (jaredlockhart) [reply app/experimenter/experiments/admin/nimbus.py:82]: Or not XD

**Pull request merged**
